### ISSUE TITLE
Momentum in Optimizers

### ISSUE THREAD

**Issue by vineeths96** [horovod/horovod#2182]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.

How is momentum implemented by Horovod? Is it based on per-replica gradient values or is it based on all-reduced gradient values?

**Comment by tgaddair** [comment]
Hey @vineeths96, sorry for the late response.  Momentum is updated with all-reduced (averaged) gradients.  This is to ensure that the model / optimizer state of each replica is consistent after every step (to avoid model parameters diverging between the workers).
